src/client/entity-editor/common/entity.js

```diff
@@ -19,49 +19,52 @@
 
 // @flow
 
-import Icon from 'react-fontawesome';
+// import Icon from 'react-fontawesome';
 import React from 'react';
 import {genEntityIconHTMLElement} from '../../helpers/entity';
 
 
 type EntityProps = {
 	disambiguation?: ?string,
-	link?: string | false,
 	text: string,
 	type: string,
 	unnamedText?: string
 };
 
 function Entity(
-	{disambiguation, link, text, type, unnamedText}: EntityProps
+	{disambiguation, id, text, type, unnamedText}: EntityProps
 ) {
-	const nameComponent = text || <i>{unnamedText}</i>;
+	const url = type === 'Area' ?
+		`//musicbrainz.org/area/${id}` :
+		`/${type.toLowerCase()}/${id}`;
+	// const nameComponent = text || <i>{unnamedText}</i>;
 	const contents = (
 		<span>
 			{
 				type && genEntityIconHTMLElement(type)
 			}
 			{' '}
-			{nameComponent}
-			{' '}
+			<a href={url} rel="noopener noreferrer" target="_blank">
+				{text} &nbsp; ( {type} )
```

Comment:
You'll still want to use the `nameComponent` instead of `text`, as it defaults to `unnamedText` if `text` is empty.

Reply:
I don't think adding the entity type in full text adds anything, considering the name is already preceded by the entity icon.
I don't think this solves BB-274, as you're not adding any other information that could help identify two entities of same type with the same name (see use-case in the ticket description). 
I think it's a bigger job, and you'll need to show different info depending on the type of entity
(see [this ticket comment](https://tickets.metabrainz.org/browse/BB-274?focusedCommentId=48269&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-48269)).

Reply:
Okay, I will try to add more info according to the entity.